parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -125,10 +125,20 @@ public class ParquetFileReader implements Closeable {
 
   public static String PARQUET_READ_PARALLELISM = "parquet.metadata.read.parallelism";
 
+  public ParquetMetricsCallback metricsCallback;
+
   private final ParquetMetadataConverter converter;
 
   private final CRC32 crc;
 
+
+  /**
+   * set a callback to send back metrics info
+   */
+  public synchronized void initMetrics(ParquetMetricsCallback callback) {
```

Comment:
why the `synchronized` is needed?

Reply:
My mistake. I initially implemented the metrics callback as a singleton. The method no longer needs to be synchronized.